### PR TITLE
Data Mock: Add Expense mock data creation script

### PR DESCRIPTION
**Pull request by shlior7**

### CODE REVIEW

File: factories/expense.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import random
+import factory
+
+from house.models import House
+from expenses.models import Expenses
+
+
+class ExpenseFactory(factory.Factory):
```
Comment (ronyyosef) [inline factories/expense.py:8]: I have found something that maybe will work better for us, try to inherit from DjangoModelFactory

File: factories/expense.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    class Meta:
+        model = Expenses
+
+    description = ""
+    house_name = factory.Iterator(House.objects.all())
+    category = factory.LazyAttribute(lambda x: random.choice(
+        [tup[1] for tup in Expenses.Category.choices]))
+    amount = factory.LazyAttribute(lambda x: random.randint(5000, 25000))
+    date = factory.Faker('date_object')
```
Comment (ronyyosef) [inline factories/expense.py:17]: I was getting an error:
RuntimeWarning: DateTimeField Expenses.date received a naive datetime (197
7-03-22 00:00:00) while time zone support is active.
  warnings.warn(

after some googling I found that the solution is to write the date field like this:
```suggestion
    date = factory.Faker('date_time', tzinfo=timezone.get_current_timezone())

```

File: scripts/create_mock_data.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from factories.expense import ExpenseFactory
+from factories.house import HouseFactory
+
+
+def run():
+    for i in range(500):
+        HouseFactory().save()
+
+    for i in range(500):
```
Comment (ronyyosef) [inline scripts/create_mock_data.py:9]: I think we can try and create more than one expense for each House, something like 5 for each house from different months.

**Comment by ronyyosef** [comment]
image

File: factories/expense.py (hunk 2 above)
Comment (shlior7) [reply factories/expense.py:17]: I knew there was a way to do that thanks man

File: scripts/create_mock_data.py (hunk 3 above)
Comment (shlior7) [reply scripts/create_mock_data.py:9]: this creates a random house that has the expense i think its not that relevant how many expenses each house has, correct me if Im wrong

Comment (ronyyosef) [reply scripts/create_mock_data.py:9]: I think that in the house(mine) view we will want to show more than one expense for each house.
Maybe even a few for each month will be better

Comment (shlior7) [reply scripts/create_mock_data.py]: changed it take a look https://github.com/redhat-beyond/Xpense/blob/a0663d56bc51cd220a044b56b32d7b396109522c/factories/expense.py#L16

Comment (ronyyosef) [reply scripts/create_mock_data.py]: Great!!
Thank you

**Review by ronyyosef — APPROVED** [review]

File: factories/expense.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @staticmethod
+    def random_date(month):
+        start_date = datetime.today().replace(day=1, month=month, tzinfo=timezone.utc)
+        end_date = ExpenseFactory.last_day_of_month(start_date)
+
+        return start_date + timedelta(
+            seconds=random.randint(0, int((end_date - start_date).total_seconds())),
+        )
+
+    house_name = factory.lazy_attribute(lambda x: x.house if x.house else House.objects.order_by('?').first())
```
Comment (adiabramovitch) [inline factories/expense.py:31]: What is the difference between ```lazy_attribute and LazyAttribute```?

**Comment by adiabramovitch** [comment]
I think you should also add some tests to show that the factory works as expected

File: factories/expense.py (hunk 4 above)
Comment (shlior7) [reply factories/expense.py:31]: I think its the same works either way, but ill change it to match the others in the method

**Comment by ronyyosef** [comment]
> I think you should also add some tests to show that the factory works as expected

I don't think we need tests for this, its not really part of the project
It will help the developers to create mock data to see how the UI looks like they want it to be look

**Pull request merged**